### PR TITLE
Add release script for external-health-monitor

### PR DESCRIPTION
**Pull request by Praveenrajmani**
This script builds and releases https://github.com/kubernetes-csi/external-health-monitor with the ubi-minimal base and including the LICENSE. This is required to certify the and publish the image in RedHat image catalog.

### CODE REVIEW

File: external-health-monitor/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.8
+
+LABEL maintainers="Kubernetes Authors"
```
Comment (balamurugana) [inline external-health-monitor/Dockerfile:3]: The maintainer is `dev@min.io` not kubernetes for this build

File: external-health-monitor/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+LABEL description="CSI External Health Monitor Controller"
+ARG binary=./bin/csi-external-health-monitor-controller
+
+COPY ${binary} csi-external-health-monitor-controller
+COPY LICENSE /licenses/LICENSE
```
Comment (balamurugana) [inline external-health-monitor/Dockerfile:8]: We have to use LICENSE file from external-health-monitor tarball.

File: external-health-monitor/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,10 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.8
+
+LABEL maintainers="Kubernetes Authors"
+LABEL description="CSI External Health Monitor Controller"
```
Comment (balamurugana) [inline external-health-monitor/Dockerfile:4]: `Rebuild of CSI External Health Monitor Controller for Red Hat Marketplace` is more appropriate

File: external-health-monitor/release.sh [hunk 4]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+# This file is part of MinIO DirectPV
+# Copyright (c) 2021, 2022, 2023 MinIO, Inc.
```
Comment (balamurugana) [inline external-health-monitor/release.sh:3]: Just `2023` here

File: external-health-monitor/release.sh [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  ${ME} <VERSION>
+
+EXAMPLE:
+  $ ${ME} 0.10.0
+EOF
+        exit 255
+    fi
+
+    BUILD_VERSION="${1/v/}"
+    IMAGE_TAG_BASE=quay.io/praveen8/csi-external-health-monitor-controller
```
Comment (balamurugana) [inline external-health-monitor/release.sh:40]: This variable is not needed. You could use the value directly like `IMG="quay.io/minio/csi-external-health-monitor-controller:v{BUILD_VERSION}"`

File: external-health-monitor/release.sh [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        PODMAN=podman
+    elif which docker >/dev/null 2>&1; then
+        PODMAN=docker
+    else
+        echo "no podman or docker found; please install"
+        exit 255
+    fi   
+}
+
+function main() {
```
Comment (balamurugana) [inline external-health-monitor/release.sh:53]: Validate whether we already released given version by doing `podman pull $IMAGE` or something

File: external-health-monitor/release.sh [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+cd "$(dirname "$0")" || exit 255
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+declare BUILD_VERSION PODMAN IMAGE_TAG_BASE IMG
+
+function init() {
+    if [ "$#" -ne 1 ]; then
```
Comment (balamurugana) [inline external-health-monitor/release.sh:28]: Make build version optional. Fetch the latest version tag of `external-health-monitor` and use it.

File: external-health-monitor/release.sh [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        PODMAN=docker
+    else
+        echo "no podman or docker found; please install"
+        exit 255
+    fi   
+}
+
+function main() {
+    # Downloading the source tar
+    curl --silent --location --insecure --fail "https://github.com/kubernetes-csi/external-health-monitor/archive/refs/tags/v${BUILD_VERSION}.tar.gz" | tar -zxf -
```
Comment (balamurugana) [inline external-health-monitor/release.sh:55]: Create a temporary build directory and do everything inside the directory. Cleanup the directory on exit.

File: external-health-monitor/release.sh [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    cd external-health-monitor-"${BUILD_VERSION}"
+    make build
+    cp ../Dockerfile cmd/csi-external-health-monitor-controller/
```
Comment (balamurugana) [inline external-health-monitor/release.sh:58]: Store $PWD and try to use absolute path with that.

File: external-health-monitor/release.sh (hunk 6 above)
Comment (Praveenrajmani) [reply external-health-monitor/release.sh:87]: currently, we have already pushed `quay.io/minio/csi-external-health-monitor-controller:v0.10.0` and now we have to update it. 

Should we delete the image and then run this script or remove this check and let the image be updated? @balamurugana

File: external-health-monitor/release.sh [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+    # Build the image
+    "${PODMAN}" buildx build --platform linux/amd64 -f cmd/csi-external-health-monitor-controller/Dockerfile --tag "${IMG}" .
+    "${PODMAN}" push "${IMG}"
+
+    # cleanup
+    cd "$work_dir" && rm -rf build
+}
+
+init "$@"
+main "$@"
```
Comment (balamurugana) [inline external-health-monitor/release.sh:106]: 1. Rename this script as `release-csi-external-health-monitor-controller.sh` and move it to project root.
2. Add below to end of `init()`
```sh
BUILD_DIR="$(mktemp -d -p "${PWD}" health-monitor-build.XXXXXXXX)"
```

3. Simplify `main()` by
```sh
curl --silent --location --insecure --fail "https://github.com/kubernetes-csi/external-health-monitor/archive/refs/tags/v${BUILD_VERSION}.tar.gz" | tar -zxf -
cd "external-health-monitor-${BUILD_VERSION}" || return
make build
"${PODMAN}" buildx build --platform linux/amd64 --tag "${IMG}" -f - . <<EOF
FROM registry.access.redhat.com/ubi8/ubi-minimal:8.8
LABEL maintainers="dev@min.io"
LABEL description="Rebuild of CSI External Health Monitor Controller for Red Hat Marketplace"
COPY ./bin/csi-external-health-monitor-controller csi-external-health-monitor-controller
COPY ./LICENSE /licenses/LICENSE
ENTRYPOINT ["/csi-external-health-monitor-controller"]
EOF
"${PODMAN}" push "${IMG}"
cd - || return
```
4. Call `main` by
```
cd "${BUILD_DIR}" || exit 255
main
cd - || exit 255
rm -fr "${BUILD_DIR}"
```
5. Remove external-health-monitor/Dockerfile file

**Review by balamurugana — APPROVED** [review]

**Pull request merged**